### ISSUE TITLE
Test for 3.16

### ISSUE THREAD

**Issue by franglais125** [franglais125/dash-isolated-3.16#1]
@nsrosenqvist
I moved here so we don't bother so much in the other issue.

So... I imported a fresh Gnome Shell 3.16 to get a compatible copy of the extension with your setup, and ported the workspace isolation implementation.

I would love to get some feedback, if you get it working!

Btw, I usually run on 3.14 (Debian stable), and sometimes work on Testing (3.20).

Cheers

**Comment by nsrosenqvist** [comment]
Hmm, when I run it with "Isolate workspace" enabled it gives me a completely empty dash. No favourites or running application windows are showing, it's just the application icon showing. When I restart the shell with it unchecked it's back to working as usual.

**Comment by franglais125** [comment]
Oh, that's too bad, sorry :(.

I'll try to make a better port tomorrow, this one was fast & loose. I'll let you know!

**Comment by nsrosenqvist** [comment]
No worries, thank you so much for working on this feature!

**Comment by franglais125** [comment]
Well, it turns out it was a typo. Can't believe myself, 876db70f1caea8f361242e51f7eed90b3f188256.

I tested it n 3.16 and it works for me!
Let me know how it goes.

**Comment by nsrosenqvist** [comment]
The feature works as it should for me. Good job!

However, when I alternate click the dash the popover opens up in the left corner and is unclickable, as opposed to right above the dash (on a dash on the bottom of screen configuration). I don't know if this bug was introduced with your code or if that is something else in the code base that has changed between 3.20 and 3.16. The extension version from extensions.gnome.org for 3.16 opens as it should above the dash.

**Comment by franglais125** [comment]
That's good to hear! If you don't mind, I have two things to ask you:

1) Do you mind getting a screenshot of the problem? I am not sure I understand what is happening.

2) To know if the problem is introduced by my changes, would you mind trying the same code I prepared, but just before the isolation commits? Here is the download link: https://github.com/franglais125/dash-isolated-3.16/archive/8171ffc3f750dccebd240f3da3cecaf3cfe785a0.zip . That zip file contains the same dash-to-dock, but without the additions I made. What's weird is that I got this copy from extensions.gnome.org, by using Gnome-Shell 3.16 on a VM.

Thanks!

**Comment by franglais125** [comment]
@nsrosenqvist So, never mind my previous comment.

I found the problem and fixed it. It was a missing line in my code. f80a3ce281e8d1473687ee71c04fd58bbf539986

Let me know if it is usable now :)

**Comment by nsrosenqvist** [comment]
Yep, it's working now!

**Comment by franglais125** [comment]
@nsrosenqvist You commented on the issue:
"I'm testing it on 3.16 and I'm loving it, works exactly as I expect it would, the overview too."

I wanted to know if you used this separate branch, or the main one?
If you managed to use the main on on 3.16, then I can delete this branch, as the main one is compatible with 3.16.

Thanks for your feedback! Glad to know it is working fine for you. If you find any bugs let me know!

**Comment by nsrosenqvist** [comment]
Sorry for the confusion, it's this branch I was testing.

**Comment by franglais125** [comment]
Not a problem, just wanted to make sure!

**Comment by franglais125** [comment]
@nsrosenqvist I'll bother you one more time :)

I ported the changes to the proper gnome-3.16 branch, so it's easier later to merge (or backport this) in the official tree, and track the commits.

Do you mind testing this branch https://github.com/franglais125/dash-to-dock/tree/gnome-3.16 ? I used it locally a bit and it seems to work properly.

Please let me know how it goes! Once you give me the OK, I'll start the pull request against micheleg's tree.

**Comment by nsrosenqvist** [comment]
I can't activate that build of the plugin. Looks like this:

![screenshot from 2016-05-28 14-02-25](https://cloud.githubusercontent.com/assets/1303475/15624500/185b5048-24dd-11e6-8e5b-c970c91811ec.png)

**Comment by franglais125** [comment]
Just to be sure, did you compile the schemas? (if you install from zip, this is sometimes necessary!)

`cd ~/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com`
`glib-compile-schemas schemas/`

You can then restart the shell, or logout and back in. If that doesn't work, then I have to check the code.

**Comment by nsrosenqvist** [comment]
Yep, I did compile the schemas. I could launch the settings but not enable the plugin. Should I try again? Any step I might have missed other than compiling the schemas?

**Comment by franglais125** [comment]
Sorry it took me so long to get back to you.
I can't really say what the problem is, but I have a couple of suggestions:
1) This new branch I mentioned has a new few commits (by micheleg), with code that hasn't been submitted yet, so maybe it's a work-in-progress. To see if the problem comes from other commits than mine, I prepared a new patch, do you mind trying it? Now, if this fails, then you have points 2) and 3), but then it's a last resort... https://github.com/franglais125/dash-to-dock/tree/gnome-3.16
(updated: https://github.com/franglais125/dash-to-dock/archive/gnome-3.16.zip).

2) Did you try this on a clean profile? e.g. a new user, so that there is no incompatibility with older settings, etc.?
3) You could try restarting the shell more than once, I had to do that a few times.

Here is hoping!

**Comment by nsrosenqvist** [comment]
Hi @franglais125, sorry for taking so long to test your changes, it's been a busy week. I tested the build you linked and after compiling the schemas it works again. I don't know if I missed restarting the shell or something the last time since it was a while ago, but at least it works now!

**Comment by franglais125** [comment]
@nsrosenqvist 
I guess I can delete this repo? It is pretty much useless as it is, as the updated branch is https://github.com/franglais125/dash-to-dock/tree/gnome-3.16.

Let me know if that's ok!

**Comment by nsrosenqvist** [comment]
@franglais125, sure, I'm using it on my system but there's no need to keep the repo. Thank you for your hard work!

**Comment by franglais125** [comment]
Perfect!

As I said, I am deleting this repo as it is a bit useless as it is (no real record tracking here).

But this branch stays alive: https://github.com/franglais125/dash-to-dock/tree/gnome-3.16

Cheers :)
